src/controllers/userController.js

```diff
@@ -0,0 +1,99 @@
+'use strict';
+
+const userService = require('../services/userService');
+
+function validateName(name, res) {
+  if (!name) {
+    res.status(400).send('Name is required');
+
+    return false;
+  }
+
+  return true;
+}
+
+async function createUser(req, res) {
+  const { name } = req.body;
+
+  if (!validateName(name, res)) {
```

Comment:
I like the idea, but this isn't the best approach.
When working with this type of validation, you can add validators to "intercept" the request before it reaches the controller, so you don't need to return a boolean for the controller to check and prevent the continuation, the validator would pause the operation before the controller. How to do it:
To declare a validator, your function `validateName` is almost  as it should be, but it needs in fact 3 arguments: req, res, next, like this:
```
function validateName(req, res, next) {
    const { name } = req.body; // same as in your controller, because it is the SAME req the controller receives

    if (!name) {
        return res.status(400)... // same code you did in your function. 
    }

    return next(); // this tells Express to continue to the next element in the flow (your controller)
}
```
Now that your validator is declared, add it to your route:
```
 router.post('/',
    validateName,
    userController.createUser
);
```
Now, if the validation fails, the controller won't even know the request has arrived, and you don't have to "double-check" the fails (first check: validation, second check: the boolean return of the validation)